components/admin_console/admin_definition.jsx

```diff
@@ -648,6 +654,306 @@ export default {
+                            needs_or: [['Enable', true], ['EnableSync', true]],
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'EmailAttribute',
+                            label: 'admin.ldap.emailAttrTitle',
+                            label_default: 'Email Attribute:',
+                            placeholder: 'admin.ldap.emailAttrEx',
+                            placeholder_default: 'E.g.: "mail" or "userPrincipalName"',
+                            help_text: 'admin.ldap.emailAttrDesc',
+                            help_text_default: 'The attribute in the AD/LDAP server that will be used to populate the email addresses of users in Mattermost.',
+                            needs_or: [['Enable', true], ['EnableSync', true]],
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'UsernameAttribute',
+                            label: 'admin.ldap.usernameAttrTitle',
+                            label_default: 'Username Attribute:',
+                            placeholder: 'admin.ldap.usernameAttrEx',
+                            placeholder_default: 'E.g.: "sAMAccountName"',
+                            help_text: 'admin.ldap.uernameAttrDesc',
+                            help_text_default: 'The attribute in the AD/LDAP server that will be used to populate the username field in Mattermost. This may be the same as the ID Attribute.',
+                            needs_or: [['Enable', true], ['EnableSync', true]],
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'IdAttribute',
+                            label: 'admin.ldap.idAttrTitle',
+                            label_default: 'ID Attribute: ',
+                            placeholder: 'admin.ldap.idAttrEx',
+                            placeholder_default: 'E.g.: "sAMAccountName"',
+                            help_text: 'admin.ldap.idAttrDesc',
+                            help_text_default: 'The attribute in the AD/LDAP server that will be used as a unique identifier in Mattermost. It should be an AD/LDAP attribute with a value that does not change, such as username or uid. If a user’s ID Attribute changes, it will create a new Mattermost account unassociated with their old one. This is the value used to log in to Mattermost in the "AD/LDAP Username" field on the sign in page. Normally this attribute is the same as the "Username Attribute" field above. If your team typically uses domain\\username to sign in to other services with AD/LDAP, you may choose to put domain\\username in this field to maintain consistency between sites.',
```

Comment:
Maybe from the old code, but should `user’s` be `user's`? (i.e. do we use the curl quotes consistently?)

Reply:
Yes, it is from the old code, but is a good catch :)

Reply:
I think @enahum can be a better idea here about this. Currently there is a lot of translation that use the `’` instead the `'`. So, I don't know if there is any reason.

Reply:
No reason at all, I believe the reason was to be able to use a single quote without escaping, but it does not make any difference really in terms of the translations

Reply:
Ok, then I'll change it in other PR for all the application.